src/vos/tests/vts_mvcc.c

```diff
@@ -306,18 +306,99 @@ fetch_f(struct io_test_args *arg, struct tx_helper *txh, char *path,
 
 static int
 fetch_dne_f(struct io_test_args *arg, struct tx_helper *txh, char *path,
-	daos_epoch_t epoch)
+	    daos_epoch_t epoch)
 {
 	return fetch_with_flags(arg, txh, path, epoch, DAOS_COND_DKEY_FETCH);
 }
 
 static int
 fetch_ane_f(struct io_test_args *arg, struct tx_helper *txh, char *path,
-	daos_epoch_t epoch)
+	    daos_epoch_t epoch)
 {
 	return fetch_with_flags(arg, txh, path, epoch, DAOS_COND_AKEY_FETCH);
 }
 
+static int
+read_ts_o(struct io_test_args *arg, struct tx_helper *txh, char *path,
+	  daos_epoch_t epoch, uint64_t flags)
+{
+	struct mvcc_arg	*mvcc_arg = arg->custom;
+	daos_unit_oid_t	 oid;
+
+	set_oid(mvcc_arg->i, path, &oid);
+
+	return tx_fetch(arg->ctx.tc_co_hdl, txh, oid, epoch, flags, NULL,
+			0, NULL, NULL);
+}
+
+static int
+read_ts_d(struct io_test_args *arg, struct tx_helper *txh, char *path,
+	  daos_epoch_t epoch, uint64_t flags)
+{
+	struct mvcc_arg	*mvcc_arg = arg->custom;
+	daos_unit_oid_t	 oid;
+	char		 dkey_buf[64];
+	daos_key_t	 dkey = {dkey_buf, sizeof(dkey_buf), 0};
+
+	set_oid(mvcc_arg->i, path, &oid);
+	set_dkey(mvcc_arg->i, path, &dkey);
+
+	return tx_fetch(arg->ctx.tc_co_hdl, txh, oid, epoch, flags, &dkey,
+			0, NULL, NULL);
+}
+
+static int
+read_ts_a(struct io_test_args *arg, struct tx_helper *txh, char *path,
```

Comment:
just one comment, rather than duplicating this function, you could have a single function that calls tx_fetch with the passed in oid and then pass in appropriate dkey and akey parameters from upper level

Reply:
in other words, change signature of read_ts_o os that it takes dkey pointer, iod_nr, and iod *...maybe change dkey func so it takes as parameter an iod_nr and iod * then you can avoid creating an akey function altogether, just create akey and pass it into the dkey function which creates a dkey and passes it (and inputs) to object function that creates an object and passes along inputs

Reply:
thanks for the feedback, I just sent a patch addressing this